src/components/global/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,60 @@
+import { ReactEventHandler, RefObject, useEffect, useRef } from "react";
+import { useDialogPolyfill } from "./useDialogPolyfill";
+import styles from "./Dialog.module.scss";
+
+export default function Dialog({
```

Comment:
I believe we should define types for props. This is one of the main goal of ts for components

Reply:
Usually for other project we name it in this style: TComponentName, here it would be:
type Dialog = {
....
}

Reply:
and also in these types we define what props are optional, and what are not